### PR TITLE
feat: [PIE-2791]: inital ACR Artifact setup

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:

Acr Artifact initial implementation

##### Jira Links:

[](https://harness.atlassian.net/browse/PIE-2791)

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts [hunk 1]
```diff
@@ -72,6 +72,10 @@ export interface ImagePathTypes {
   repository?: string
   repositoryUrl?: string
   repositoryPortorRepositoryURL?: string
+  dockerRepositoryServer?: string
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:75]: It's better to define a separate type for AcrArtifact. Similarly we can separate out for each artifact later.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        ? `pipeline.stages.${stageIdentifier}.spec.serviceConfig.stageOverrides.artifacts.${artifactPath}.spec.tag`
+        : `pipeline.stages.${stageIdentifier}.spec.serviceConfig.serviceDefinition.spec.artifacts.${artifactPath}.spec.tag`
+    },
+    lazy: true
+  })
+
+  useEffect(() => {
+    if (fromPipelineInputTriggerTab(formik, fromTrigger) && !isEmpty(formik?.values?.selectedArtifact)) {
+      if (isSidecar) {
+        setSidecarInitialValues(initialValues, formik, stageIdentifier, artifactPath as string)
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:118]: This is not needed any more. This is moved to Kubernetesprimary and sidecar artifacts. Please rebase your branch

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx [hunk 3]
```diff
@@ -4,12 +4,572 @@
...
+      values.registry = registries.find(registry => registry.value === specValues?.registry)?.value as string
+    }
+    if (getMultiTypeFromValue(specValues?.repository) === MultiTypeInputType.FIXED) {
+      values.repository = repositories.find(repository => repository.value === specValues?.repository)?.value as string
+    }
+    return values
+  }, [context, initialValues, subscriptions, registries, repositories, selectedArtifact])
+
+  const submitFormData = (formData: ImagePathTypes & { connectorId?: string }): void => {
+    const artifactObj = getFinalArtifactObj(formData, context === 2)
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx:266]: We have enum defined for this in ArtifactHelper. `context === ModalViewFor.SIDECAR`

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx [hunk 4]
```diff
@@ -4,12 +4,572 @@
...
+                    },
+                    allowableTypes
+                  }}
+                  label={getString('common.cluster')}
+                />
+                {getMultiTypeFromValue(formik.values.registry) === MultiTypeInputType.RUNTIME && !isReadonly && (
+                  <ConfigureOptions
+                    value={formik.values.registry || ''}
+                    type="String"
+                    variableName="registry"
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx:451]: variableName is wrong here. Check for all ConfigureOptions

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx (hunk 3 above)
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx:266]: changed, thanksss! :)

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx (hunk 4 above)
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AcrArtifact/AcrArtifact.tsx:451]: fixed, thaaanks!

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx (hunk 2 above)
Comment (Tamaraius) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:118]: removed, thanks!

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts (hunk 1 above)
Comment (Tamaraius) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactInterface.ts:75]: created seperate ACRArtifactType, thanks!

**Pull request merged**